ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/Substring.java

```diff
@@ -16,23 +16,22 @@
 
 package io.confluent.ksql.function.udf.string;
 
-import io.confluent.ksql.function.KsqlFunctionException;
-import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
 
-public class SubstringKudf implements Kudf {
+@UdfDescription(name = "substring",
+    author = "Confluent",
+    description = "returns a substring of the passed in value")
+public class Substring {
 
-  @Override
-  public Object evaluate(Object... args) {
-    if ((args.length < 2) || (args.length > 3)) {
-      throw new KsqlFunctionException("Substring udf should have two or three input argument.");
-    }
-    String string = args[0].toString();
-    int start = (Integer) args[1];
-    if (args.length == 2) {
-      return string.substring(start);
-    } else {
-      int end = (Integer) args[2];
-      return string.substring(start, end);
-    }
+  @Udf
+  public String substring(final String value, final int start) {
```

Comment:
Should we also have descriptions for individual functions?

Reply:
we can do. I'll add it. thanks

Reply:
+1, I wonder how fancy you can get with these annotations, but in an ideal world it would be  would be good to be able to enforce documentation of the argument types, and also the semantics of the function. Perhaps the best we can do is set good examples by documenting the internal functions nicely.

Reply:
Also, it is so nice to see properly typed methods!

Reply:
proper typing is great! can't wait to use it...
Also, fancier enforcement, typing, and docs with annotations is totally possible - check this example, from https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

`   @Description("suffix starting at given index")`
`   @ScalarFunction`
`   @LiteralParameters("x")`
`   @SqlType("varchar(x)")`
`    public static Slice substr(@SqlType("varchar(x)") Slice utf8, @SqlType(StandardTypes.BIGINT) long start)
    {
`

Would love to see something like this one day